Storage/src/HmacKey.php

```diff
@@ -0,0 +1,217 @@
+ */
+
+namespace Google\Cloud\Storage;
+
+use Google\Cloud\Storage\Connection\ConnectionInterface;
+
+/**
+ * Represents a Service Account HMAC key.
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\Storage\StorageClient;
+ *
+ * $storage = new StorageClient();
+ * $hmacKey = $storage->hmacKey($accessId);
+ * ```
+ */
+class HmacKey
+{
+    /**
+     * @var ConnectionInterface
+     */
+    private $connection;
+
+    /**
+     * @var string
+     */
+    private $projectId;
+
+    /**
+     * @var string
+     */
+    private $accessId;
+
+    /**
+     * @var array|null
+     */
+    private $info;
+
+    /**
+     * @var string|null
+     */
+    private $secret;
+
+    /**
+     * @param ConnectionInterface $connection A connection to Cloud Storage.
+     * @param string $projectId The current project ID.
+     * @param string $accessId The key identifier.
+     * @param array|null $info The key metadata.
+     * @param string|null $secret The key secret.
+     */
+    public function __construct(
+        ConnectionInterface $connection,
+        $projectId,
+        $accessId,
+        array $info = [],
+        $secret = null
+    ) {
+        $this->connection = $connection;
+        $this->projectId = $projectId;
+        $this->accessId = $accessId;
+        $this->info = $info;
+        $this->secret = $secret;
+    }
+
+    /**
+     * Get the HMAC Key Access ID.
+     *
+     * Example:
+     * ```
+     * $accessId = $hmacKey->accessId();
+     * ```
+     *
+     * @return string
+     */
+    public function accessId()
+    {
+        return $this->accessId;
+    }
+
+    /**
+     * Fetch the key metadata from Cloud Storage.
+     *
+     * Example:
+     * ```
+     * $keyMetadata = $hmacKey->reload();
+     * ```
+     *
+     * @param array $options {
+     *     Configuration Options
+     *
+     *     @type string $userProject If set, this is the ID of the project which
+     *           will be billed for the request.
+     * }
+     * @return array
+     */
+    public function reload(array $options = [])
+    {
+        $this->info = $this->connection->getHmacKey([
+            'projectId' => $this->projectId,
+            'accessId' => $this->accessId
+        ] + $options);
+
+        return $this->info;
+    }
+
+    /**
+     * Get the HMAC Key Metadata.
+     *
+     * If the metadata is not already available, it will be requested from Cloud
+     * Storage.
+     *
+     * Example:
+     * ```
+     * $keyMetadata = $hmacKey->info();
+     * ```
+     *
+     * @param array $options {
+     *     Configuration Options
+     *
+     *     @type string $userProject If set, this is the ID of the project which
+     *           will be billed for the request.
+     * }
+     * @return array
+     */
+    public function info(array $options = [])
+    {
+        return $this->info ?: $this->reload($options);
+    }
+
+    /**
+     * Get the HMAC Key Secret.
+     *
+     * Only populated immediately after key creation.
```

Comment:
I don't think we should store the returned secret in the runtime for security reasons. It should only be available when the user calls createHmacKey for the first time.

@frankyn thought here?